plugin/src/main/java/dev/regadas/trino/pubsub/listener/pubsub/Publisher.java

```diff
@@ -0,0 +1,10 @@
+package dev.regadas.trino.pubsub.listener.pubsub;
+
+import com.google.protobuf.Message;
+import java.util.concurrent.CompletableFuture;
+
+public interface Publisher {
```

Comment:
Here we can extend the `Closeable` removing the need to expose directly the shutdown.